contrib/labstack/echo.v4/echotrace.go

```diff
@@ -83,6 +83,11 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// serve the request to the next middleware
 			err := next(c)
 			if err != nil {
+				// if ignoreError is not set, or if error should not be ignored,
+				// tag the span with error
+				if cfg.errCheck == nil || !cfg.errCheck(err) {
+					finishOpts = append(finishOpts, tracer.WithError(err))
+				}
```

Comment:
```suggestion
			if err != nil && (cfg.errCheck == nil || cfg.errCheck(err)) {
```

Since the logic adding `tracer.WithError(err)` was already here (lines 99 and and 105).

Also, there are 2 more cases where the span could be marked as an error, with the custom errors we are creating from the response status code (lines 111 and 116). Maybe it would make sense to call this function there as well, to give control over these errors. What do you think?

Reply:
yeah I figured as well, but I think the logic is a bit tangled up... what do you think the priority should be? as a matter of fact, I think this `errCheck` option should supersede that. I initially thought of providing that when I started this PR, but then I think we should give user more control rather than just the status code.

Reply:
see my tests. I was wanting the users themselves to do the `errors.As` and compare the code there, rather than only providing the code. they could even `errors.As` into a custom error.

Reply:
as for the branches when there's no error but different code... I don't know, how would the users use that? compare the error string?

Reply:
what do you think of this @rarguelloF ? how should we untangle this, i.e., the logic of this error check and the returned status code?